src/core/network.js

```diff
@@ -414,9 +415,27 @@ PDFNetworkStreamFullRequestReader.prototype = {
       networkManager.abortRequest(fullRequestXhrId);
     }
 
+    // Content-Disposition: attachment; filename=Naïve file.txt
+    var fullRequestXhr = networkManager.getRequestXhr(fullRequestXhrId);
+    var contentDisposition =
+      fullRequestXhr.getResponseHeader('Content-Disposition');
+    this._fileName = this._extractNameFrom(contentDisposition);
+
     this._headersReceivedCapability.resolve();
   },
 
+  _extractNameFrom:
+      function PDFNetworkStreamFullRequestReader_extractNameFrom(cDisposition) {
+    if (cDisposition) {
+      var parts = /.+; filename="(.+)"/gi.exec(cDisposition);
```

Comment:
Good morning @Snuffleupagus 

I've already added it but I need some assistance from somebody with more experience than me:

I've been also working in the branch conflicts. I've discovered that I made some newbie mistakes during the branch process: Basically I didn't executed this step:

git remote add upstream git://github.com/mozilla/pdf.js.git

I tried to do it on the branch for this pull request and it resulted in a small mess.

Then I decided to start again, following the steps in the contribution  guide and adding all the changes to a new branch.

Now I have everything working, in just one commit.

What I don't know how to do is replace the invalid branch of this pull request  with the right one. How can I do it? Or should I create a new pull request?

PD: I'm not pushed the changes yet.